src/node_url_pattern.cc

```diff
@@ -123,13 +123,13 @@ URLPattern::URLPattern(Environment* env,
 }
 
 void URLPattern::MemoryInfo(MemoryTracker* tracker) const {
-  tracker->TrackFieldWithSize("protocol", url_pattern_.get_protocol().size());
-  tracker->TrackFieldWithSize("username", url_pattern_.get_username().size());
-  tracker->TrackFieldWithSize("password", url_pattern_.get_password().size());
-  tracker->TrackFieldWithSize("hostname", url_pattern_.get_hostname().size());
-  tracker->TrackFieldWithSize("pathname", url_pattern_.get_pathname().size());
-  tracker->TrackFieldWithSize("search", url_pattern_.get_search().size());
-  tracker->TrackFieldWithSize("hash", url_pattern_.get_hash().size());
+  tracker->TrackField("protocol", url_pattern_.get_protocol());
+  tracker->TrackField("username", url_pattern_.get_username());
+  tracker->TrackField("password", url_pattern_.get_password());
+  tracker->TrackField("hostname", url_pattern_.get_hostname());
+  tracker->TrackField("pathname", url_pattern_.get_pathname());
+  tracker->TrackField("search", url_pattern_.get_search());
+  tracker->TrackField("hash", url_pattern_.get_hash());
```

Comment:
IMO, the memory info of `node::URLPattern` should report the size of `ada::url_pattern<URLPatternRegexProvider> pattern_`, which is a field of `node::URLPattern`, rather than the indirect internal fields of `ada::url_pattern`.

Reply:
```suggestion
  tracker->TrackFieldWithSize("url_pattern_", sizeof(url_pattern_));
```

Reply:
Actually I think it needs to be both the individual fields and the struct given that the size of the struct won't report the size of the underlying string allocations.